packages/pwa/app/commerce-api/index.js

```diff
@@ -201,11 +202,16 @@ class CommerceAPI {
             await this.auth.login()
         }
 
+        const url = new URL(getAppOrigin())
```

Comment:
Why did you choose to construct `url` from `getAppOrigin`?

In the context of this method, that is not the URL that will be called.

I believe in the context of this function, you have access to the real URL being requested – let's use that!